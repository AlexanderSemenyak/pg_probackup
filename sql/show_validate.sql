-- test show command
\! rm -rf ${PWD}/results/sample_backup
\! cp -rp data/sample_backup ${PWD}/results/sample_backup
\! pg_arman show -A ${PWD}/results/arclog -B ${PWD}/results/sample_backup
\! pg_arman validate -B ${PWD}/results/sample_backup 2009-05-31 17:05:53 --debug
\! pg_arman validate -B ${PWD}/results/sample_backup 2009-06-01 17:05:53 --debug
\! pg_arman show -a -A ${PWD}/results/arclog -B ${PWD}/results/sample_backup
\! pg_arman show 2009-06-01 17:05:53 -A ${PWD}/results/arclog -B ${PWD}/results/sample_backup
