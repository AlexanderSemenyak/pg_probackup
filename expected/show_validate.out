-- test show command
\! rm -rf ${PWD}/results/sample_backup
\! cp -rp data/sample_backup ${PWD}/results/sample_backup
\! pg_arman show -A ${PWD}/results/arclog -B ${PWD}/results/sample_backup
==========================================================================
Start                Mode  Current TLI  Parent TLI  Time    Data  Status  
==========================================================================
2009-06-03 17:05:53  FULL            1           0    0m    ----  RUNNING 
2009-06-01 17:05:53  INCR            1           0    3m  9223PB  DONE 
2009-05-31 17:05:53  FULL            1           0    3m  1242MB  DONE 
\! pg_arman validate -B ${PWD}/results/sample_backup 2009-05-31 17:05:53 --debug
INFO: validate: 2009-05-31 17:05:53 backup and archive log files by CRC
LOG: database files...
LOG: (1/1) PG_VERSION
LOG: backup 2009-05-31 17:05:53 is valid
\! pg_arman validate -B ${PWD}/results/sample_backup 2009-06-01 17:05:53 --debug
INFO: validate: 2009-06-01 17:05:53 backup and archive log files by CRC
LOG: database files...
LOG: (1/1) PG_VERSION
WARNING: CRC of backup file "PG_VERSION" must be 0 but FEF71BC1
WARNING: backup 2009-06-01 17:05:53 is corrupted
\! pg_arman show -a -A ${PWD}/results/arclog -B ${PWD}/results/sample_backup
==========================================================================
Start                Mode  Current TLI  Parent TLI  Time    Data  Status  
==========================================================================
2009-06-03 17:05:53  FULL            1           0    0m    ----  RUNNING 
2009-06-02 17:05:03  FULL            1           0    0m    ----  DELETED 
2009-06-01 17:05:53  INCR            1           0    3m  9223PB  CORRUPT 
2009-05-31 17:05:53  FULL            1           0    3m  1242MB  OK 
\! pg_arman show 2009-06-01 17:05:53 -A ${PWD}/results/arclog -B ${PWD}/results/sample_backup
# configuration
BACKUP_MODE=INCREMENTAL
COMPRESS_DATA=false
# result
TIMELINEID=1
START_LSN=0/0b40c800
STOP_LSN=0/0b4c8020
START_TIME='2009-06-01 17:05:53'
END_TIME='2009-06-01 17:09:13'
RECOVERY_XID=0
DATA_BYTES=9223372036854775807
BLOCK_SIZE=8192
XLOG_BLOCK_SIZE=8192
STATUS=CORRUPT
